### DISCUSSION TITLE
VPN secure Gluetun latest image issues

### DISCUSSION THREAD

**Discussion by JohnTa15** [qdm12/gluetun#1453] [Q&A] [answered]
Hello, 

I'm having an issue with gluetun. I use VPNsecure as a VPN and I follow your instructions [https://github.com/qdm12/gluetun/wiki/VPN-Secure](url) but it is not working. After finishing your steps I deploy the container and then I see those errors

> 2023-03-13T20:57:06Z INFO [openvpn] OpenSSL: error:0308010C:digital envelope routines::unsupported
> 2023-03-13T20:57:06Z INFO [openvpn] OpenSSL: error:0308010C:digital envelope routines::unsupported
> 2023-03-13T20:57:06Z INFO [openvpn] Cannot load private key file [[INLINE]]
> 2023-03-13T20:57:06Z INFO [openvpn] SIGUSR1[soft,private-key-password-failure] received, process restarting
 
Also here is my gluetun's directory. The client.crt and openvpn_encrypted_key are downloaded from vpnsecure site. 

> drwxr-xr-x  2 root root    4096 Mar 13 22:32 .
> drwxr-xr-x 14 root root    4096 Mar 13 21:16 ..
> -rw-r--r--  1 root root    1086 Mar 13 21:17 client.crt
> -rw-r--r--  1 root root    1058 Mar 13 21:17 openvpn_encrypted_key
> -rw-r--r--  1 root root 3533102 Mar 13 22:54 servers.json

Finally here is my compose file(I have replaced abc with my user and pass):

> version: '3'
> services:
>   gluetun:
>     image: qmcgaw/gluetun
>     container_name: gluetun
>     restart: unless-stopped
>     environment:
>       - VPN_SERVICE_PROVIDER=vpnsecure
>       - OPENVPN_USER=abc
>       - OPENVPN_PASSWORD=abc
>       - SERVER_REGIONS=Hesse
>     volumes:
>       - /portainer/Files/AppData/Config/gluetun:/gluetun
>     cap_add:
>       - NET_ADMIN

**Comment by cllu** [comment]
For me, the issue is that the newer version of OpenSSL doesn't support old/insecure algorithms, which is what VPNSecure uses to generate the certificates. I copied the `/etc/ssl/openssl.cnf` file out, edited it according to [this post](https://stackoverflow.com/a/74416513) and then mount the modified file in the same path. It now works!

**Reply by qdm12** [reply]
I'm not sure on the indirect security implications of modifying openssl behavior. I am working on a fix in #1451 hopefully it can be automated, and in the worst case you'll have to run an openssl command to decrypt+re-encrypt the key they provide you. Honestly I don't know what the hell vpn secure is doing with des encrypted keys. They might as well not encrypt them at all. And also it's so easy to just change the encryption scheme for generating keys, they are disappointing 😢

**Comment by JohnTa15** [comment]
@cllu actually that solved my issue https://github.com/qdm12/gluetun/issues/1451#issuecomment-1468114137

**Reply by qdm12** [reply]
This is not a permanent solution though, I'm working on fixing the latest image (probably decrypt the DES key and re-encrypt it using AES, automatically).

**Comment by qdm12** [comment, marked as answer]
@JohnTa15 @cllu the latest image and v3.33.0 image both now transparently upgrade the encryption of the key if it's DES-CBC encrypted, hence solving the problem you had.

**Reply by cllu** [reply]
Thanks @qdm12, the latest image works perfectly!
